src/components/Book.js

```diff
@@ -1,29 +1,35 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { removeBooks } from '../redux/books/booksSlice';
 
-const Book = ({ book, onDelete }) => {
-  const handleDelete = () => {
-    onDelete(book.id);
-  };
+const Book = ({ id, title, author }) => {
+  const dispatch = useDispatch();
 
   return (
     <div>
-      <h3>{book.title}</h3>
-      <p>{book.author}</p>
-      <button type="button" onClick={handleDelete}>
+      <h3>{title}</h3>
+      <p>{author}</p>
+      <button
+        type="button"
+        onClick={() => {
+          dispatch(removeBooks(id));
+        }}
+      >
         Delete
       </button>
```

Comment:
- [ ] @Mahdi-Niazi Great work using redux in your components, however, the delete button doesn't work on newly added books, this is because you are passing the id instead of item_id that you are using in your booksSlice to delete a book.